sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -154,15 +154,8 @@ object FileSourceStrategy extends Strategy with PredicateHelper with Logging {
         l.resolve(
           fsRelation.partitionSchema, fsRelation.sparkSession.sessionState.analyzer.resolver)
       val partitionSet = AttributeSet(partitionColumns)
-      val partitionKeyFilters = if (partitionColumns.isEmpty) {
-        ExpressionSet(Nil)
-      } else {
-        val predicates = ExpressionSet(normalizedFilters
-          .filter(_.references.subsetOf(partitionSet)))
-        logInfo(s"Pruning directories with: ${predicates.mkString(",")}")
-        predicates
-      }
-
+      val partitionKeyFilters = DataSourceStrategy.getPushedDownFilters(partitionColumns,
```

Comment:
Nit: It would be better to add a comment saying that the `partitionKeyFilters`  here is the same as the filters used in the partition pruning rule `PruneFileSourcePartitions`.